File: src/test/java/com/github/ccguyka/showupdates/ArtifactVersionListBuilder.java
package com.github.ccguyka.showupdates;

import java.util.ArrayList;
import java.util.List;

import org.apache.maven.artifact.versioning.ArtifactVersion;
import org.apache.maven.artifact.versioning.DefaultArtifactVersion;

public class ArtifactVersionListBuilder {

    private final List<ArtifactVersion> versions = new ArrayList<>();

    private ArtifactVersionListBuilder() {
        // prevent instantiation
    }

    public static ArtifactVersionListBuilder updates() {
        return new ArtifactVersionListBuilder();
    }

    public ArtifactVersionListBuilder version(String version) {
        versions.add(new DefaultArtifactVersion(version));

        return this;
    }

    public List<ArtifactVersion> build() {
        return versions;
    }

}

File: src/test/kotlin/com/github/ccguyka/showupdates/ArtifactVersionListBuilder.kt
package com.github.ccguyka.showupdates

import org.apache.maven.artifact.versioning.ArtifactVersion
import org.apache.maven.artifact.versioning.DefaultArtifactVersion
import java.util.*

class ArtifactVersionListBuilder private constructor() {

    private val versions: MutableList<ArtifactVersion> = ArrayList()

    fun version(version: String?): ArtifactVersionListBuilder {
        versions.add(DefaultArtifactVersion(version))
        return this
    }

    fun build(): List<ArtifactVersion> {
        return versions
    }

    companion object {
        @JvmStatic
        fun updates(): ArtifactVersionListBuilder {
            return ArtifactVersionListBuilder()
        }
    }
}
